src/prefect/cli/worker.py

```diff
@@ -244,6 +264,9 @@ async def _get_worker_class(
     if worker_type is None:
         worker_type = await _retrieve_worker_type_from_pool(work_pool_name)
 
+    if worker_type == "prefect-agent":
+        return None
+
```

Comment:
To provide better guidance to users, I suggest moving this check into the main body of the `start` command and exit with instructions for the user like so:
```
if worker_type == "prefect-agent":
     exit_with_error("'prefect-agent' typed work pools work with Prefect Agents instead of Workers. Please use the 'prefect agent start' to start a Prefect Agent.")
```

Reply:
@desertaxle I did declare those lines in the 'start' function, but the error is still persisting. It's only working fine if I declare these lines in the '_get_worker_class' function, and it works fine in that case, I guess.

Reply:
You can add a test in this file: https://github.com/PrefectHQ/prefect/blob/main/tests/cli/test_worker.py

Reply:
You can invoke the CLI directly in a test and assert that the exit code is 1 and that the expected message is printed. You can look at `test_install_policy_prompt_decline` in this file as an example.

Reply:
@desertaxle  Can you check the logic now and tell the required changes ?